test/unit/commands/plugin/install.test.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const chai = require('chai');
+const sinon = require('sinon');
+const yaml = require('js-yaml');
+const path = require('path');
+const fse = require('fs-extra');
+const proxyquire = require('proxyquire');
+const fixturesEngine = require('../../../fixtures/programmatic');
+const resolveConfigurationPath = require('../../../../lib/cli/resolve-configuration-path');
+const { expect } = require('chai');
+
+chai.use(require('chai-as-promised'));
+
+const npmCommand = 'npm';
+
+describe('test/unit/commands/plugin/install.test.js', async () => {
+  let spawnFake;
+  let serviceDir;
+  let configurationFilePath;
+
+  const pluginName = 'serverless-plugin-1';
+
+  before(async () => {
+    spawnFake = sinon.fake(async (command, args) => {
+      if (command === npmCommand && args[0] === 'install' && args[1] === '--save-dev') {
+        const _pluginName = args[2];
+        const pluginNameWithoutVersion = _pluginName.split('@')[0];
+
+        if (pluginNameWithoutVersion) {
+          const pluginPackageJsonFilePath = path.join(
+            serviceDir,
+            'node_modules',
+            pluginName,
+            'package.json'
+          );
+          const packageJsonFileContent = {};
+          await fse.ensureFile(pluginPackageJsonFilePath);
+          await fse.writeJson(pluginPackageJsonFilePath, packageJsonFileContent);
```

Comment:
Do we need this logic?

Reply:
Otherwise, there will be an error here:
https://github.com/serverless/serverless/blob/d04f05f27117978b105b08cd4215622fc96c4188/commands/plugin/install.js#L123

I saw you mentioned that this step (`installPeerDependencies`) would not be necessary as it's done by npm. But you might have removed the comment?

If the `installPeerDependencies` step is needed, then we would need this empty `package.json`.

Let me know @medikoo

Reply:
I didn't remove the comment (not sure what happened to it :).

I believe it's ok to skip peer dependencies install